### ISSUE TITLE
Raspberry Pi to ESP8266

### ISSUE THREAD

**Issue by dbourdeveloper** [not-matt/Systematic-LEDs#70]
So far I've seen issues related to using the PI itself to generate the data, but what if I wanted to connect my ESP8266's to my Pi. How do I change the audio input/output similar to the voice meter on windows?

The ideal setup I have is
(1) Set up Pi so my phone can connect to it through bluetooth
(2) Play music on my phone
(3) Music sends signal to my ESP8266s

Would this be feasible or too under powered?

**Comment by dbourdeveloper** [comment]
> Did u have any issues running main.py in RPI? I'm trying to running on Linux but a have a bunch of error, I would like to know if u have any error running this in raspian

What kind of errors are you getting? I did have to modify the main.py a bit to function correctly and can share my fixes. It's not entirely perfect but it works.

**Comment by richyxi** [comment]
Sure, thanks man, I just came for work I'll try it in a sec

**Comment by dbourdeveloper** [comment]
> hi, it works good, thanks man, but it seems like its a little unestable when im trying to change the mic set up, same with the color control, and first i was playing with the effects options and it crashed, all of this things happend to your raspberry? also, im curious of how you can connect your phone to the RPI and play music? in fact, im working on this project and trying to do the same as you. cheers
> 
> pd. if someone is facing the same issue i modify main.py with those lines of code #71 and #53

Glad it works. There's some still GUI functions that crash that I haven't gotten around to fixing. Is this the issues you're referring to? Is it something along the lines of a `KeyError`? 

As for the Raspberry Pi, I used bluetooth to connect my phone to my Pi, then ran the systematic-led script. Here's a tutorial of how you can activate bluetooth (https://didier.io/post/raspberrypi-stream-sound-over-bluetooth) Note, it will be very laggy and not sure if this approach is worth it.

**Comment by richyxi** [comment]
Yeah that was with the effects option, but sometimes happen and sometimes
it goes well,  a little wierd, do you think it only happens with
Linux based systems??

What if a aux cable is used trough a USB audio device?

**Comment by dbourdeveloper** [comment]
The aux might help with the lag a bit as its one less layer, but I think the main issue is just the Pi isn't powerful/fast enough to output the data to many ESP8266s at once.

**Comment by richyxi** [comment]
Maybe with some pi alternatives like these, may have different results 
https://www.zdnet.com/pictures/top-12-raspberry-pi-alternatives-february-2019/3/
